beacon-chain/sync/decode_pubsub.go

```diff
@@ -51,7 +57,19 @@ func (s *Service) decodePubsubMessage(msg *pubsub.Message) (ssz.Unmarshaler, err
 	}
 	// Handle different message types across forks.
 	if topic == p2p.BlockSubnetTopicFormat {
-		m, err = extractBlockDataType(fDigest[:], s.cfg.clock)
+		m, err = extractDataType(types.BlockMap, fDigest[:], s.cfg.clock)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if topic == p2p.AttestationSubnetTopicFormat {
+		m, err = extractDataType(types.AttestationMap, fDigest[:], s.cfg.clock)
```

Comment:
instead of passing a map for each type, we could just pass the topic and the method does the rest.

Reply:
but then the function would no longer be generic and the same code would have to be duplicated for each map

Reply:
You infer the type from the topic

Reply:
ex: an attestation topic will always use the attestation map and never anything else. This switching is done in the `extractDataType` method itself